samples/sample-07/src/test/java/com/example/sample07/NewConsumerRebalanceProtocolTest.java

```diff
@@ -0,0 +1,325 @@
+			return executor;
+		}
+
+		@Bean
+		GenericContainer genericContainer(ConsumerFactory<Integer, String> cf) throws InterruptedException {
+			GenericContainer genericContainer = setUpBroker();
+
+			Properties props = new Properties();
+			props.setProperty(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG, BytesDeserializer.class.getName());
+			props.setProperty(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG,
+							  StringDeserializer.class.getName());
+			props.setProperty(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, BROKER);
+			props.setProperty(ConsumerConfig.GROUP_ID_CONFIG, "hello");
+			props.setProperty(ConsumerConfig.GROUP_PROTOCOL_CONFIG, "consumer");
+			props.setProperty(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest");
+			Consumer<Object, Object> dummyConsumer = new KafkaConsumer<>(props);
+
+			Thread.sleep(5000);
+			dummyConsumer.close();
+
+			return genericContainer;
+		}
+
+		GenericContainer setUpBroker() {
+			final DockerImageName imageName = DockerImageName.parse(KAFKA_IMAGE_NAME);
+			final GenericContainer broker = new GenericContainer(imageName);
+
+			// Set KRAFT.
+			broker.addEnv("KAFKA_CFG_NODE_ID", "0");
+			broker.addEnv("KAFKA_CFG_KRAFT_CLUSTER_ID", "HsDBs9l6UUmQq7Y5E6bNlw");
+			broker.addEnv("KAFKA_CFG_CONTROLLER_QUORUM_VOTERS", "0@localhost:9093");
+			broker.addEnv("KAFKA_CFG_PROCESS_ROLES", "controller,broker");
+
+			// Set listener.
+			broker.addEnv("KAFKA_CFG_LISTENERS",
+						  "INTERNAL://localhost:29092, PLAINTEXT://0.0.0.0:9092, EXTERNAL://:9094, CONTROLLER://:9093");
+			broker.addEnv("KAFKA_CFG_ADVERTISED_LISTENERS",
+						  "INTERNAL://localhost:29092, PLAINTEXT://localhost:9092, EXTERNAL://127.0.0.1:12000");
+			broker.addEnv("KAFKA_CFG_LISTENER_SECURITY_PROTOCOL_MAP",
+						  "CONTROLLER:PLAINTEXT,EXTERNAL:PLAINTEXT,PLAINTEXT:PLAINTEXT,INTERNAL:PLAINTEXT");
+			broker.addEnv("KAFKA_CFG_CONTROLLER_LISTENER_NAMES", "CONTROLLER");
+
+			// Set ETC.
+			broker.addEnv("KAFKA_CFG_AUTO_CREATE_TOPICS_ENABLE", "true");
+			broker.addEnv("KAFKA_CFG_NUM_PARTITIONS", "2");
+			broker.addEnv("KAFKA_CFG_INTER_BROKER_LISTENER_NAME", "INTERNAL");
+			broker.addEnv("KAFKA_CFG_GROUP_INITIAL_REBALANCE_DELAY_MS", "0");
+			broker.addEnv("KAFKA_CFG_TRANSACTION_STATE_LOG_MIN_ISR", "1");
+			broker.addEnv("KAFKA_CFG_TRANSACTION_STATE_LOG_REPLICATION_FACTOR", "1");
+			broker.addEnv("KAFKA_CFG_OFFSETS_TOPIC_REPLICATION_FACTOR", "1");
+			broker.addEnv("KAFKA_CFG_DEFAULT_REPLICATION_FACTOR", "1");
+
+			// KIP-848. If you want more, see links below.
+			// https://cwiki.apache.org/confluence/display/KAFKA/KIP-848%3A+The+Next+Generation+of+the+Consumer+Rebalance+Protocol
+			// https://cwiki.apache.org/confluence/display/KAFKA/The+Next+Generation+of+the+Consumer+Rebalance+Protocol+%28KIP-848%29+-+Early+Access+Release+Notes
+			broker.addEnv("KAFKA_CFG_GROUP_COORDINATOR_REBALANCE_PROTOCOLS", "classic,consumer");
+			broker.addEnv("KAFKA_CFG_TRANSACTION_PARTITION_VERIFICATION_ENABLE", "false");
```

Comment:
Cannot all of these env vars be extracted into a properties files?
And then we can use `broker.getEnvMap().putAll()`